### PR TITLE
[MM 40414] : Update empty states to align with new branding and changes in empty state copy

### LINKED ISSUE #19325: Webapp: Update empty states to align with new branding
The current empty states need to be updated to align with the new branding. Below are the proposed visuals. Minor copy updates to be made as well - [see the Figma file for details](https://www.figma.com/file/2YwSsBAEYbmKmhPjWflxBo/?node-id=0%3A1)

Messaging for these empty states have also been updated.

### RHS Empty States

image

### Channel Switcher Modal Empty State

image

### Emoji Picker Popover

image

### Links

[https://www.figma.com/file/2YwSsBAEYbmKmhPjWflxBo/?node-id=0%3A1](https://www.figma.com/file/2YwSsBAEYbmKmhPjWflxBo/?node-id=0%3A1|smart-link) 

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-40414

### PR DESCRIPTION
**Pull request by VishakhaPoonia**
#### Summary
Update empty states to align with new branding and changes in empty state copy

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/19325
  Fixes https://mattermost.atlassian.net/browse/MM-40414

#### Related Pull Requests
n/a

#### Screenshots

Screenshot 2022-04-04 at 12 53 54 AM
Screenshot 2022-04-04 at 12 55 16 AM
Screenshot 2022-04-04 at 12 53 43 AM
Screenshot 2022-04-04 at 12 53 35 AM
Screenshot 2022-04-04 at 12 53 26 AM
Screenshot 2022-04-04 at 12 53 17 AM
Screenshot 2022-04-04 at 12 52 59 AM
Screenshot 2022-04-04 at 12 53 08 AM
Screenshot 2022-04-25 at 12 43 26 PM
Screenshot 2022-04-25 at 12 44 18 PM

#### Release Note
```release-note
UI: Empty state icons changes and related text copy changes.
```

### CODE REVIEW

**Comment by esethna** [comment]
@VishakhaPoonia thanks for the PR! Is it ready for review?

**Comment by VishakhaPoonia** [comment]
@esethna No there are few changes left will try to close those today. Thanks

**Comment by VishakhaPoonia** [comment]
@esethna Pr is ready for review, Thanks for waiting.

**Review by larkox — APPROVED** [review]
LGTM, just two comments not completely related with the PR.

File: components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap [hunk 1]
```diff
@@ -22,7 +22,7 @@ exports[`components/global/AtMentionsButton should match snapshot 1`] = `
   }
 >
   <ForwardRef
-    aria-label="Select to toggle a list of saved posts."
+    aria-label="Select to toggle a list of Saved Messages."
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap:25]: Yeah, they should be. There's even a comment in the code saying that they should be localized, so I guess that was forgotten when the global header got merged. I'll file a ticket to follow up on those

File: components/widgets/icons/channel_files_icon.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+export default function ChannelFilesIcon(props: React.HTMLAttributes<HTMLSpanElement>) {
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply components/widgets/icons/channel_files_icon.tsx:7]: I'm not sure. Maybe @matthewbirtch would know?

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Great work @VishakhaPoonia. I have a few very minor tweaks:

1. Can we make sure the body text on all the empty states is updated to use `color: rgba(var(--center-channel-color-rgb), .72);`?
2. For the search results empty state, can we use proper `“` and `”` characters for the search term?
3. For the emoji picker empty state, can we hide "Select an Emoji" from the footer when the results are empty?

Other than that, this looks great!

File: components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (VishakhaPoonia) [reply components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap:25]: @hmhealey Do I need to do something or I can mark it as resolved.

File: components/widgets/icons/channel_files_icon.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (VishakhaPoonia) [reply components/widgets/icons/channel_files_icon.tsx:7]: @larkox @hmhealey Do you folks mind sharing the link?

**Comment by VishakhaPoonia** [comment]
> 2. For the search results empty state, can we use proper `“` and `”` characters for the search term?
@matthewbirtch Are you mentioning quote style, I checked in Figma you are using `Metropolis` font, in code we are using the same font. I can think of 2 scenario’s 

1. Either only name is same and font used here and in Figma are different,
2. Font isn’t getting downloaded.

Screenshot 2022-04-05 at 9 16 54 AM

**Comment by matthewbirtch** [comment]
> > 2. For the search results empty state, can we use proper `“` and `”` characters for the search term?
> 
> @matthewbirtch Are you mentioning quote style, I checked in Figma you are using `Metropolis` font, in code we are using the same font. I can think of 2 scenarios
> 
> 1. Either only name is same and font used here and in Figma are different,
> 2. Font isn’t getting downloaded.

The font is correct - that's not the problem. You just need to update the quote characters to change from using `"` to proper opening and closing quote marks `“` and `”`

File: components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (larkox) [reply components/global_header/right_controls/saved_posts_button/__snapshots__/saved_posts_button.test.tsx.snap:25]: No actions from your side @VishakhaPoonia . Thanks!

File: components/widgets/icons/channel_files_icon.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hmhealey) [reply components/widgets/icons/channel_files_icon.tsx:7]: Oh yeah, that's a good point. If that's the case with this image, it seems like we put them in `components/common/svg_images_components` currently, and we name them accordingly

**Review by matthewbirtch — APPROVED** [review]
Awesome work @VishakhaPoonia. Thanks for the contribution!

**Comment by esethna** [comment]
Adding @AshishDhama for a second dev review

File: components/emoji_picker/emoji_picker.tsx [hunk 3]
```diff
@@ -1,18 +1,18 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useRef, useState, useEffect, useCallback, memo, useMemo} from 'react';
-import {FormattedMessage} from 'react-intl';
-import type {FixedSizeList} from 'react-window';
+import React, { useRef, useState, useEffect, useCallback, memo, useMemo } from 'react';
```
Comment (hmhealey) [inline components/emoji_picker/emoji_picker.tsx:4]: Could you revert the changes your autoformatter made to the spacing inside the curly brackets in this file? That's causing the linter to fail, and it might also be the cause of some of the conflicts in this file

**Comment by VishakhaPoonia** [comment]
@hmhealey Have updated the Pr, please take look.

**Review by AshishDhama — APPROVED** [review]
LGTM!

**Review by larkox — APPROVED** [review]

**Comment by hmhealey** [comment]
I haven't taken a super deep look on this, but I'll trust @larkox and @AshishDhama's judgment here

**Comment by esethna** [comment]
cc// @jgilliam17 for QA review

**Comment by jgilliam17** [comment]
Thanks @VishakhaPoonia 
Tested manually and changes look good.
One question for UX @matthewbirtch the new graphic in dark themes doesn't show the icon "shadow" as it does in light theme. Is that ok?
Screen Shot 2022-04-25 at 4 26 27 PM
Screen Shot 2022-04-25 at 4 26 12 PM

Quite a few [E2Es ](https://mattermost-cypress-report.s3.amazonaws.com/187327-1a5a57e-onprem-ent-webapp-pr-10040-mm-ee-test:1a5a57e/mochawesome.html) are failing due to wording changes pinned posts > pinned messages, save > save message etc. Can you please take a look and fix the failing tests? Thanks 🙂

**Comment by matthewbirtch** [comment]
> One question for UX @matthewbirtch the new graphic in dark themes doesn't show the icon "shadow" as it does in light theme. Is that ok?

Yes, @jgilliam17 this is expected.

**Comment by stylianosrigas** [comment]
Removing cloud test server for maintenance reasons. Please feel free to add it again if needed.

**Comment by jgilliam17** [comment]
@VishakhaPoonia Let me know if you need any help with [E2Es](https://mattermost-cypress-report.s3.amazonaws.com/187327-1a5a57e-onprem-ent-webapp-pr-10040-mm-ee-test:1a5a57e/mochawesome.html) that failed due to the changes in this PR.

**Comment by esethna** [comment]
@VishakhaPoonia let us know if you're still working on this?

**Comment by esethna** [comment]
@VishakhaPoonia are you able to help with teh E2E's? This PR is really close, I'd love to get it across the finish line!

**Comment by VishakhaPoonia** [comment]
@esethna Yeah I will take care of this weekend, Thanks for the reminder.

**Comment by nevyangelova** [comment]
@VishakhaPoonia I will handle the E2e tests so we can merge this this week. Really appreciate the work you did for this feature 💯

**Comment by VishakhaPoonia** [comment]
@nevyangelova thanks @nevyangelova for the help, I do have some progress either I will create a new pr or update this one by tomorrow. If that will work.

**Comment by nevyangelova** [comment]
> @nevyangelova thanks @nevyangelova for the help, I do have some progress either I will create a new pr or update this one by tomorrow. If that will work.

ok we can wait no worries.

**Comment by rmonga71** [comment]
> @nevyangelova thanks @nevyangelova for the help, I do have some progress either I will create a new pr or update this one by tomorrow. If that will work.

@VishakhaPoonia thanks for the great work - please keep us posted on this!

**Comment by VishakhaPoonia** [comment]
@nevyangelova have updated test cases, if you can take a look I am very new to e2e

**Comment by nevyangelova** [comment]
@VishakhaPoonia thanks for updating. I see some quotes being inserted by accident it seems? Also @saturninoabril can confirm if the tests look good.

For unit tests that are failing you can run ```npm test -- -u```

**Comment by jgilliam17** [comment]
Thanks @VishakhaPoonia 
Few tests are still failing. See full report [here](https://mattermost-cypress-report.s3.amazonaws.com/220345-c701238-onprem-ent-webapp-pr-10040-mattermost/mm-ee-testc701238/mochawesome.html) 
Here are few spec files that need updating, this may not be a complete list - please check the linked report.

`e2e/cypress/tests/integration/messaging/tooltips_on_top_nav_channel_icons_posts_spec.js`

`e2e/cypress/tests/integration/accessibility/accessibility_buttons_spec.js`

`e2e/cypress/tests/integration/search/clear_input_spec.js`

`e2e/cypress/tests/integration/search/cleared_search_term_spec.js`

For test case MM-T4801, I think the change here appears to be accidental 
We should continue to display “Save” on post menu in mobile view and not "Save Message" @matthewbirtch  can you please confirm
Screen Shot 2022-07-27 at 4 48 02 PM

**Comment by matthewbirtch** [comment]
> We should continue to display “Save” on post menu in mobile view and not "Save Message" @matthewbirtch can you please confirm

@jgilliam17 yes it should just be "Save", but I don't think this has anything to do with this PR though does it?

**Comment by jgilliam17** [comment]
Thanks @matthewbirtch - Just checking if this was an intentional change. I agree this is not within scope of this PR, but it seems to have been accidentally changed during text updates.

**Review by rmonga71 — APPROVED** [review]
Looks good to me, thanks for the contribution @VishakhaPoonia!

**Comment by nevyangelova** [comment]
@VishakhaPoonia some minor conflicts need to be resolved otherwise this is ready to go!

**Comment by VishakhaPoonia** [comment]
@nevyangelova Thanks will take care of this tomorrow.

**Comment by esethna** [comment]
@VishakhaPoonia how's this going? :)

**Comment by AshishDhama** [comment]
@jgilliam17 @esethna It’s ready for testing

**Comment by VishakhaPoonia** [comment]
@jgilliam17 Have resolved the conflicts. Could you please help me get this merged?

**Comment by jgilliam17** [comment]
Thanks @VishakhaPoonia 
Few tests are failing, see full [report here](https://mattermost-cypress-report.s3.amazonaws.com/246975-db5cdc7-onprem-ent-webapp-pr-10040-mm-ee-test:db5cdc7/mochawesome.html)
If you need help with Cypress please ask here or reach out in [QA channel ](https://community.mattermost.com/core/channels/quality-assurance)
Here are all cypress test cases that need to be updated with the new text:
MM-T1461 aria label should be updated to Pinned Messages
MM-T1722 test should look for "Select to toggle a list of Saved Messages" 
MM-T3659 test should look for "Select to toggle a list of Saved Messages" 
MM-T4059_5 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T4801 Change the menu item back to "Save" it is currently failing because the menu has been changed to "Save Message"
MM-T200 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T202 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T2170 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T2171 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T122 test should look for accessible element "Select to toggle a list of Saved Messages" 
MM-T3352  (MM-T3353 & MM-T3354 in the same set failing) update saved posts to Saved Messages
MM-T4948_1 change back to Save for post menu item, we're not changing to Save Message 
MM-T4948_2 update test text to Saved Messages 
MM-T134 update test text for tooltip to Pinned Messages 
MM-T368 test should look for accessible element "Select to toggle a list of Saved Messages" 
Handle removed user- new sidebar test - test should look for accessible element "Select to toggle a list of Saved Messages"

**Comment by esethna** [comment]
@VishakhaPoonia are you able to help with teh above test updates? ^

**Comment by esethna** [comment]
Adding @Rishselva to help shepherd this in

**Comment by Rishselva** [comment]
Hello @VishakhaPoonia! A gentle ping to see if there are any new updates.

**Comment by matthewbirtch** [comment]
Hi @VishakhaPoonia just checking in on this. I think the only thing left was addressing the E2E failing tests. @jgilliam17 can you help with this?

**Comment by jgilliam17** [comment]
@VishakhaPoonia Can you please rebase branch with master so I can trigger e2es again? 
If you have nay questions about my previous [post](https://github.com/mattermost/mattermost-webapp/pull/10040#issuecomment-1283949215) let me know. Thanks 🙂

**Comment by AshishDhama** [comment]
@matthewbirtch Yeah only test are left, I will fix those and get this merged

**Comment by matthewbirtch** [comment]
@AshishDhama did we get the failing tests fixed? Did we need to reassign a QA team member here?

**Comment by AshishDhama** [comment]
@matthewbirtch I will finish it next week, nothing much except e2e

**Review by furqanmlk — COMMENTED** [review]
@VishakhaPoonia 
1. `No File yet` text is not correct.

image

2. `No groups yet` and `No members yet` model does not have text aligned

image



image

**Comment by AshishDhama** [comment]
> @VishakhaPoonia
> 
> 1. `No File yet` text is not correct.
> 
> image
> 
> 2. `No groups yet` and `No members yet` model does not have text aligned
> 
> image
> 
> image

@furqanmlk I think 2 and 3 should be fine, but refer to the UX team for that @matthewbirtch @asaadmahmood 
for 1. I would refer to @cwarnermm

**Comment by cwarnermm** [comment]
The Figma text for #1 has a different tone than the others in the series. Proposing we change "You'll see files here when someone attaches a file to a post in this channel." to "You'll see files here when someone shares files in this channel."

@justinegeffen - Would love your input on this too.

**Comment by justinegeffen** [comment]
Thanks, @cwarnermm. Is this empty state for all channels including DMs and GMs? If so, something more general might work better:

"You'll see files here when they're shared in this channel."

**Comment by cwarnermm** [comment]
Love your suggestion, @justinegeffen. @VishakhaPoonia - can you confirm that this text would apply to all 4 channel types, including direct messages and group messages, please?

**Comment by AshishDhama** [comment]
@cwarnermm Yeah, it will be applied to all 4 cases.

**Comment by hmhealey** [comment]
The web app now lives in the `webapp/channels` over in https://github.com/mattermost/mattermost-server. We decided to move it to be a monorepo to make both development and our CI pipelines easier, but it's unfortunately disruptive right now. We've updated [our setup docs](https://developers.mattermost.com/contribute/developer-setup/) and added [some steps for migrating to working with the monorepo](https://developers.mattermost.com/contribute/monorepo-migration-notes/).

Copying the changes over there hopefully shouldn't be too hard. There's a few changes to file paths (such as the code living in a dedicated `src` subdirectory), but most things should look the same as they do now. 

Let us know if you could use any help with this. Hopefully the process is smooth.

**Comment by hmhealey** [comment]
The web app now lives in the `webapp/channels` over in https://github.com/mattermost/mattermost-server. We decided to move it to be a monorepo to make both development and our CI pipelines easier, but it's unfortunately disruptive right now. We've updated [our setup docs](https://developers.mattermost.com/contribute/developer-setup/) and added [some steps for migrating to working with the monorepo](https://developers.mattermost.com/contribute/monorepo-migration-notes/).

Copying the changes over there hopefully shouldn't be too hard. There's a few changes to file paths (such as the code living in a dedicated `src` subdirectory), but most things should look the same as they do now. 

Let us know if you could use any help with this. Hopefully the process is smooth.
